cmd/query/main.go

```diff
@@ -126,25 +126,7 @@ func parseopts() opts {
 	return opts
 }
 
-// @title        VDS-slice API
-// @version      0.0
-// @description  Serves seismic slices and fences from VDS files.
-// @contact.name Equinor ASA
-// @contact.url  https://github.com/equinor/vds-slice/issues
-// @license.name GNU Affero General Public License
-// @license.url  https://www.gnu.org/licenses/agpl-3.0.en.html
-// @schemes      https
-func main() {
-	opts := parseopts()
-
-	storageAccounts := strings.Split(opts.storageAccounts, ",")
-	
-	endpoint := api.Endpoint{
-		MakeVdsConnection: vds.MakeAzureConnection(storageAccounts),
-		Cache:             cache.NewCache(opts.cacheSize),
-	}
-
-	app := gin.New()
+func setupRouting(app *gin.Engine, endpoint *api.Endpoint, opts *opts) {
```

Comment:
This comment is a bit misplaced due to GitHub.

I don't think setting up  and running the metrics app fits in this function. This function, as the name suggests, should only be responsible for adding the right routes and middleware to the main app. 

You could make another function for the metrics app, (Essentially everything in the `if opts.metrics` expect the 2 first lines), but I would just inline that in main unless we need the same setup for testing. 

Also I would prefer that we pass in the variables from opts that we need, not the entire opts struct

Reply:
Mostly agree, but ended up with a bit different version. I don't exactly like it, but I can't pinpoint what I dislike and how to fix it, so this seems to me the best option.